dali/base/dasds.cpp

```diff
@@ -798,12 +798,7 @@ class CSubscriberContainer : public CSubscriberContainerBase
                     if (numeric)
                     {
                         unsigned qnum = atoi(qualifier);
-                        if (!item.queryParent())
-                        {
-                            if (qnum != 1)
-                                return false;
-                        }
-                        else if (((PTree *)item.queryParent())->findChild(&item) != qnum-1)
+                        if (qnum != item.querySiblingPosition())
```

Comment:
Had to change because queryParent changed, but change here is really incidental, to improve readability by using a new querySiblingPosition() method.